code/game/turfs/simulated/minerals.dm

```diff
@@ -59,11 +59,20 @@
 		to_chat(user, "<span class='notice'>You start picking...</span>")
 		P.playDigSound()
 
-		if(do_after(user,P.digspeed, target = src))
-			if(ismineralturf(src))
-				to_chat(user, "<span class='notice'>You finish cutting into the rock.</span>")
-				gets_drilled(user)
-				SSblackbox.add_details("pick_used_mining","[P.type]")
+		var/obj/item/organ/alcoholvessel/dwarf //check to see if the digger has the dwarf organ for increased mining speed
+		dwarf = user.getorganslot("dwarf_organ")
+		if(!dwarf)
+			if(do_after(user,P.digspeed, target = src)) //regular mining slowdown
+				if(ismineralturf(src))
+					to_chat(user, "<span class='notice'>You finish cutting into the rock.</span>")
+					gets_drilled(user)
+					SSblackbox.add_details("pick_used_mining","[P.type]")
+		else
+			if(do_after(user,P.digspeed * 0.5, target = src)) //50% of normal slowdown, dwarf speed.
+				if(ismineralturf(src))
+					to_chat(user, "<span class='notice'>You finish cutting into the rock.</span>")
+					gets_drilled(user)
+					SSblackbox.add_details("pick_used_mining","[P.type]")
```

Comment:
I suggest you just add something like  `var/adjusted_digspeed = (user.getorganslot("dwarf_organ") ? (P.digspeed * 0.5) : P.digspeed)` and then just change the do_after. You then will not have to duplicate the check, and I don't see why you defined dwarf as an organ, I think it could have just been an ordinary variable, but maybe I am wrong.